sdk/core/System.Net.ClientModel/tests/client/MapsClient/MapsClient.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.ClientModel;
+using System.Net.ClientModel.Core;
+using System.Text;
+using System.Threading;
+
+namespace Maps;
+
+public class MapsClient
+{
+    private readonly Uri _endpoint;
+    private readonly KeyCredential _credential;
+    private readonly MessagePipeline _pipeline;
+    private readonly string _apiVersion;
+
+    public MapsClient(Uri endpoint, KeyCredential credential, MapsClientOptions options = default)
+    {
+        if (endpoint is null) throw new ArgumentNullException(nameof(endpoint));
+        if (credential is null) throw new ArgumentNullException(nameof(credential));
+
+        options ??= new MapsClientOptions();
+
+        _endpoint = endpoint;
+        _credential = credential;
+        _apiVersion = options.Version;
+
+        if (options.PerCallPolicies is null)
+        {
+            options.PerCallPolicies = new PipelinePolicy[1];
+        }
+        else
+        {
+            var perCallPolicies = new PipelinePolicy[options.PerCallPolicies.Length + 1];
+            options.PerCallPolicies.CopyTo(perCallPolicies.AsSpan());
+        }
+
+        options.PerCallPolicies[options.PerCallPolicies.Length - 1] = new KeyCredentialAuthenticationPolicy(_credential, "subscription-key");
```

Comment:
I wonder if we should consider being able to decorate `PipelinePolicy` implementation types with interfaces like `ISupportsPerTry` so that it's easier to know where a policy belongs in the policy chain. For example, many auth policies are pre-try because they require per-request signing, but this one happens to be ok for perCall. It would be nice if policies helped you fall into the pit of success with regard to how they are inteded to be used and positioned.

Reply:
Yeah, at some point we had DefaultLocation property on HttpPolicy. You could then do Add(policy) and it would go to its default location. Pavel removed it for some reason.

Reply:
I'm working on pipeline options now in https://github.com/Azure/azure-sdk-for-net/pull/39740, and this will change as part of that.

It seems like smartly inferring the position of a policy is something we could add later if needed?  @christothes do you see anything that would require us to plan in advance for this feature?

Reply:
If we wanted to put a property or method on `PipelinePolicy` to support this, we'd need to do it sooner.